crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -51,6 +51,13 @@ pub struct AuthorityEpochTables<S> {
     /// epoch change.
     pub(crate) consensus_message_processed: DBMap<TransactionDigest, bool>,
 
+    /// This is an inverse index for consensus_message_processed - it allows to select
+    /// all transactions at the specific consensus range
+    ///
+    /// The consensus position for the transaction is defined as first position at which valid
+    /// certificate for this transaction is seen in consensus
+    pub(crate) consensus_message_order: DBMap<ExecutionIndices, TransactionDigest>,
```

Comment:
it seems we have a similar table in Narwhal, do we plan to remove it from narwhal and use this one instead?

Reply:
We actually don't have similar table in Narwhal (not anymore anyway).
Executor in Narwhal is now completely stateless, any commit-related state that we preserve is ConsensusOutput order.